File: api_nestjs_distributed_order_platform/src/presentation/http/mappers/client.mapper.ts
import { ClientEntity } from "src/domain/entities/client.entity";

export function toClientResponse(client: ClientEntity) {
    return {
        id: client.getId(),
        name: client.getName(),
        email: client.getEmail(),
        status: client.getStatus(),
        createdAt: client.getCreatedAt(),
        updatedAt: client.getUpdatedAt(),
    };
}

File: api_java_distributed_order_platform/src/main/java/com/project/order/presentation/http/mapper/ClientMapper.java
package com.project.order.presentation.http.mapper;

import com.project.order.domain.model.Client;
import com.project.order.presentation.http.dto.ClientDto.ClientResponse;

public class ClientMapper {

    private ClientMapper() {}

    public static ClientResponse toResponse(Client client) {
        return new ClientResponse(
                client.getId(),
                client.getName(),
                client.getEmail().getValue(),
                client.getStatus().name(),
                client.getCreatedAt(),
                client.getUpdatedAt()
        );
    }
}
